alphastats/DataSet.py

```diff
@@ -127,6 +178,15 @@ def _remove_misc_samples_in_metadata(self):
                 "are removed from the metadata."
             )
 
+    # TODO this is implemented in both preprocessing and here
```

Comment:
To fix the todo: This is only needed in the DimensionalityReduction class and only if the step was not run during preprocessing. I would suggest replacing the step in DimensionalityReduction with something like:
`mat = self.data.mat.loc[sample_names,:]` after creating sample_names.

Reply:
added this as a comment